MsWheaPkg/Crates/RustMuTelemetryHelperLib/src/lib.rs

```diff
@@ -0,0 +1,263 @@
+//! Rust MU Telemetry Helper
+//!
+//! Rust helper library for logging telemetry.
+//!
+//! ## Examples and Usage
+//!
+//! ```no_run
+//! use rust_mu_telemetry_helper_lib::{init_telemetry, log_telemetry};
+//! use r_efi::{efi, system};
+//! pub extern "efiapi" fn efi_main(
+//!     _image_handle: efi::Handle,
+//!     system_table: *const system::SystemTable,
+//!  ) -> efi::Status {
+//!
+//!    //Initialize Boot Services
+//!    unsafe {
+//!        init_telemetry((*system_table).boot_services.as_ref().unwrap());
+//!    }
+//!
+//!    //if (some_failure) {
+//!        let _ = log_telemetry(false, 0xA1A2A3A4, 0xB1B2B3B4B5B6B7B8, 0xC1C2C3C4C5C6C7C8, None, None, None);
+//!    //}
+//!
+//!    efi::Status::SUCCESS
+//! }
+//! ```
+//!
+//! ## License
+//!
+//! Copyright (C) Microsoft Corporation.
+
+//!
+//! SPDX-License-Identifier: BSD-2-Clause-Patent
+//!
+#![cfg_attr(target_os = "uefi", no_std)]
+
+mod status_code_runtime;
+
+use boot_services::{BootServices, StandardBootServices};
+use mu_pi::protocols::status_code::{EfiStatusCodeType, EfiStatusCodeValue};
+use mu_pi::status_code::{EFI_ERROR_CODE, EFI_ERROR_MAJOR, EFI_ERROR_MINOR};
+use mu_rust_helpers::{guid, guid::guid};
+use r_efi::efi;
+use status_code_runtime::{ReportStatusCode, StatusCodeRuntimeProtocol};
+
+static BOOT_SERVICES: StandardBootServices = StandardBootServices::new_uninit();
+
+/// Matches gMsWheaRSCDataTypeGuid in MsWheaPkg\MsWheaPkg.dec
+/// Matches MS_WHEA_RSC_DATA_TYPE in MsWheaPkg\Private\Guid\MsWheaReportDataType.h
+const MS_WHEA_RSC_DATA_TYPE_GUID: efi::Guid = guid!("91DEEA05-8C0A-4DCD-B91E-F21CA0C68405");
+
+const MS_WHEA_ERROR_STATUS_TYPE_INFO: EfiStatusCodeType = EFI_ERROR_MINOR | EFI_ERROR_CODE;
+const MS_WHEA_ERROR_STATUS_TYPE_FATAL: EfiStatusCodeType = EFI_ERROR_MAJOR | EFI_ERROR_CODE;
+
+/**
+ Internal RSC Extended Data Buffer format used by Project Mu firmware WHEA infrastructure.
+
+ A Buffer of this format should be passed to ReportStatusCodeWithExtendedData
+
+ LibraryID:         GUID of the library reporting the error. If not from a library use zero guid
+ IhvSharingGuid:    GUID of the partner to share this with. If none use zero guid
+ AdditionalInfo1:   64 bit value used for caller to include necessary interrogative information
+ AdditionalInfo2:   64 bit value used for caller to include necessary interrogative information
+**/
+// #pragma pack(1)
+// typedef struct {
+//     EFI_GUID    LibraryID;
+//     EFI_GUID    IhvSharingGuid;
+//     UINT64      AdditionalInfo1;
+//     UINT64      AdditionalInfo2;
+//   } MS_WHEA_RSC_INTERNAL_ERROR_DATA;
+// #pragma pack()
+
+#[repr(C)]
+struct MsWheaRscInternalErrorData {
+    library_id: efi::Guid,
+    ihv_sharing_guid: efi::Guid,
+    additional_info_1: u64,
+    additional_info_2: u64,
+}
+
+/// Log telemetry

```

Comment:
The `is_fatal` parameter is missing in the comment.

Reply:
Looks like this was missing in the original library as well. Added parameter and updated all argument descriptors to snake case.